lib/dependencies/RequireEnsureDependency.js

```diff
@@ -2,24 +2,30 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var NullDependency = require("./NullDependency");
-var DepBlockHelpers = require("./DepBlockHelpers");
+"use strict";
+const NullDependency = require("./NullDependency");
+const DepBlockHelpers = require("./DepBlockHelpers");
 
-function RequireEnsureDependency(block) {
-	NullDependency.call(this);
-	this.block = block;
-}
-module.exports = RequireEnsureDependency;
+class RequireEnsureDependency extends NullDependency {
+	constructor(block) {
+		super();
+		this.block = block;
+	}
 
-RequireEnsureDependency.prototype = Object.create(NullDependency.prototype);
-RequireEnsureDependency.prototype.constructor = RequireEnsureDependency;
-RequireEnsureDependency.prototype.type = "require.ensure";
+	get type() {
+		return "require.ensure";
```

Comment:
@Kovensky 
>defining `this.type` in the constructor

This will create `this.type` for every instance

> attaching it to the RequireEnsureDependency.prototype after the class definition

👍